medusa/storage/google_storage.py

```diff
@@ -127,18 +127,15 @@ async def _upload_object(self, data: io.BytesIO, object_key: str, headers: t.Dic
             )
         )
 
-        storage_class = self.get_storage_class()
-        ex_header = {"storageClass": storage_class} if storage_class else {}
         resp = await self.gcs_storage.upload(
             bucket=self.bucket_name,
             object_name=object_key,
             file_data=data,
             force_resumable_upload=True,
             timeout=-1,
-            headers=ex_header,
         )
         return AbstractBlob(
-            resp['name'], int(resp['size']), resp['md5Hash'], resp['timeCreated'], storage_class.upper()
+            resp['name'], int(resp['size']), resp['md5Hash'], resp['timeCreated'], None
```

Comment:
In the other two providers, we use the information returned by the upload operation. In GCS case, we don't get this info, so we're not setting it at all.